### PR TITLE
SABR formula for in-arrears overnight cap/floor (Asian style options)

### PR DESCRIPTION
**Pull request by marc-henrard**
SABR effective parameters for in-arrears cap/floors.
Using formulas from Willems, Sander. SABR smiles for RFR caplets. August 2020.  Electronic copy available at: https://ssrn.com/abstract=3567655 (Theorem 4.1 and Theorem 4.2)

### CODE REVIEW

File: modules/pricer/src/main/java/com/opengamma/strata/pricer/impl/volatility/smile/SabrInarrearsVolatilityFunction.java [hunk 1]
```diff
@@ -0,0 +1,629 @@
...
+    double h = nu * nu * (tau2 + 2 * q * tau02 + tau12) / (2 * tau1 * tau * (q + 1)) - nuHat2;
+    double alphaHat2 = alpha * alpha / (2 * q + 1) * tau / tau1 * Math.exp(0.5 * h * tau1);
+    double alphaHat = Math.sqrt(alphaHat2);
+    return SabrFormulaData.of(alphaHat, beta, rhoHat, nuHat);
+  }
+
+  /**
+   * The effective SABR parameters from the raw SABR parameters and the times. Formula in the case tau0 <= 0.
+   * <p>
+   * Theorem 4.2 in reference.
```
Comment (benhamilton-brown) [inline modules/pricer/src/main/java/com/opengamma/strata/pricer/impl/volatility/smile/SabrInarrearsVolatilityFunction.java:94]:
```suggestion
   * Theorem 4.1 in reference.
```

File: modules/pricer/src/main/java/com/opengamma/strata/pricer/impl/volatility/smile/SabrInarrearsVolatilityFunction.java [hunk 2]
```diff
@@ -0,0 +1,629 @@
...
+   * @param tau1  the accumulation period end time
+   * @return  the effective SABR parameters
+   */
+  public SabrFormulaData effectiveSabrBeforeStart(SabrFormulaData parameters, double tau0, double tau1) {
+    double alpha = parameters.getAlpha();
+    double beta = parameters.getBeta();
+    double rho = parameters.getRho();
+    double nu = parameters.getNu();
+    double tau = 2 * q * tau0 + tau1;
+    double tau2 = tau * tau;
```
Comment (benhamilton-brown) [inline modules/pricer/src/main/java/com/opengamma/strata/pricer/impl/volatility/smile/SabrInarrearsVolatilityFunction.java:71]: Maybe use underscore to show it's squared i.e tau1_2 and tau0_2. Do you think its clearer what is subscript and what is a power?

**Review by benhamilton-brown — CHANGES_REQUESTED** [review]
I've reviewed the formulas, I didn't go through the AD thoroughly but have made the assumption that your fd tests should confirm they are right

File: modules/pricer/src/main/java/com/opengamma/strata/pricer/impl/volatility/smile/SabrInarrearsVolatilityFunction.java (hunk 1 above)
Comment (marc-henrard) [reply modules/pricer/src/main/java/com/opengamma/strata/pricer/impl/volatility/smile/SabrInarrearsVolatilityFunction.java:94]: Done

File: modules/pricer/src/main/java/com/opengamma/strata/pricer/impl/volatility/smile/SabrInarrearsVolatilityFunction.java (hunk 2 above)
Comment (marc-henrard) [reply modules/pricer/src/main/java/com/opengamma/strata/pricer/impl/volatility/smile/SabrInarrearsVolatilityFunction.java:71]: Good idea, as we have tau1 and tau2 with different meaning of the number.

Comment (marc-henrard) [reply modules/pricer/src/main/java/com/opengamma/strata/pricer/impl/volatility/smile/SabrInarrearsVolatilityFunction.java]: Actually, stylecheck does not like the '_' in the names. I reverted to the old names.

**Review by benhamilton-brown — APPROVED** [review]

**Pull request merged**